### PR TITLE
Update all subworkflows to new NF syntax

### PR DESCRIPTION
**Pull request by drpatelh**

### CODE REVIEW

File: tests/subworkflows/nf-core/gatk_create_somatic_pon/main.nf [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+            [ file(params.test_data['homo_sapiens']['illumina']['test2_paired_end_recalibrated_sorted_bam'], checkIfExists: true) ],
+            [ file(params.test_data['homo_sapiens']['illumina']['test2_paired_end_recalibrated_sorted_bam_bai'], checkIfExists: true) ],
+            []
+        ]
+    ]
+    fasta         = file(params.test_data['homo_sapiens']['genome']['genome_fasta'], checkIfExists: true)
+    fai           = file(params.test_data['homo_sapiens']['genome']['genome_fasta_fai'], checkIfExists: true)
+    dict          = file(params.test_data['homo_sapiens']['genome']['genome_dict'], checkIfExists: true)
+    pon_name      = "test_panel"
+    interval_file = file(params.test_data['homo_sapiens']['genome']['genome_interval_list'], checkIfExists: true)
```
Comment (GCJMackenzie) [inline tests/subworkflows/nf-core/gatk_create_somatic_pon/main.nf:26]:
```suggestion
    fasta         = file(params.test_data['homo_sapiens']['genome']['genome_21_fasta'], checkIfExists: true)
    fai           = file(params.test_data['homo_sapiens']['genome']['genome_21_fasta_fai'], checkIfExists: true)
    dict          = file(params.test_data['homo_sapiens']['genome']['genome_21_dict'], checkIfExists: true)
    pon_name      = "test_panel"
    interval_file = file(params.test_data['homo_sapiens']['genome']['genome_21_interval_list'], checkIfExists: true)
```

File: tests/subworkflows/nf-core/gatk_tumor_normal_somatic_variant_calling/main.nf [hunk 2]
```diff
@@ -2,24 +2,31 @@
...
+        ]
+    ]
+    fasta                 = file(params.test_data['homo_sapiens']['genome']['genome_fasta'], checkIfExists: true)
+    fai                   = file(params.test_data['homo_sapiens']['genome']['genome_fasta_fai'], checkIfExists: true)
+    dict                  = file(params.test_data['homo_sapiens']['genome']['genome_dict'], checkIfExists: true)
+    germline_resource     = file(params.test_data['homo_sapiens']['genome']['gnomad_r2_1_1_vcf_gz'], checkIfExists: true)
+    germline_resource_tbi = file(params.test_data['homo_sapiens']['genome']['gnomad_r2_1_1_vcf_gz_tbi'], checkIfExists: true)
+    panel_of_normals      = file(params.test_data['homo_sapiens']['genome']['mills_and_1000g_indels_vcf_gz'], checkIfExists: true)
+    panel_of_normals_tbi  = file(params.test_data['homo_sapiens']['genome']['mills_and_1000g_indels_vcf_gz_tbi'], checkIfExists: true)
+    interval_file         = file(params.test_data['homo_sapiens']['genome']['genome_interval_list'], checkIfExists: true)
```
Comment (GCJMackenzie) [inline tests/subworkflows/nf-core/gatk_tumor_normal_somatic_variant_calling/main.nf:29]:
```suggestion
    fasta                 = file(params.test_data['homo_sapiens']['genome']['genome_21_fasta'], checkIfExists: true)
    fai                   = file(params.test_data['homo_sapiens']['genome']['genome_21_fasta_fai'], checkIfExists: true)
    dict                  = file(params.test_data['homo_sapiens']['genome']['genome_21_dict'], checkIfExists: true)
    germline_resource     = file(params.test_data['homo_sapiens']['genome']['gnomad_r2_1_1_21_vcf_gz'], checkIfExists: true)
    germline_resource_tbi = file(params.test_data['homo_sapiens']['genome']['gnomad_r2_1_1_21_vcf_gz_tbi'], checkIfExists: true)
    panel_of_normals      = file(params.test_data['homo_sapiens']['genome']['mills_and_1000g_indels_21_vcf_gz'], checkIfExists: true)
    panel_of_normals_tbi  = file(params.test_data['homo_sapiens']['genome']['mills_and_1000g_indels_21_vcf_gz_tbi'], checkIfExists: true)
    interval_file         = file(params.test_data['homo_sapiens']['genome']['genome_21_interval_list'], checkIfExists: true)
```

File: tests/subworkflows/nf-core/gatk_tumor_only_somatic_variant_calling/main.nf [hunk 3]
```diff
@@ -2,23 +2,25 @@
...
+        ]
+    ]
+    fasta                 = file(params.test_data['homo_sapiens']['genome']['genome_fasta'], checkIfExists: true)
+    fai                   = file(params.test_data['homo_sapiens']['genome']['genome_fasta_fai'], checkIfExists: true)
+    dict                  = file(params.test_data['homo_sapiens']['genome']['genome_dict'], checkIfExists: true)
+    germline_resource     = file(params.test_data['homo_sapiens']['genome']['gnomad_r2_1_1_vcf_gz'], checkIfExists: true)
+    germline_resource_tbi = file(params.test_data['homo_sapiens']['genome']['gnomad_r2_1_1_vcf_gz_tbi'], checkIfExists: true)
+    panel_of_normals      = file(params.test_data['homo_sapiens']['genome']['mills_and_1000g_indels_vcf_gz'], checkIfExists: true)
+    panel_of_normals_tbi  = file(params.test_data['homo_sapiens']['genome']['mills_and_1000g_indels_vcf_gz_tbi'], checkIfExists: true)
+    interval_file         = file(params.test_data['homo_sapiens']['genome']['genome_interval_list'], checkIfExists: true)
```
Comment (GCJMackenzie) [inline tests/subworkflows/nf-core/gatk_tumor_only_somatic_variant_calling/main.nf:23]:
```suggestion
    fasta                 = file(params.test_data['homo_sapiens']['genome']['genome_21_fasta'], checkIfExists: true)
    fai                   = file(params.test_data['homo_sapiens']['genome']['genome_21_fasta_fai'], checkIfExists: true)
    dict                  = file(params.test_data['homo_sapiens']['genome']['genome_21_dict'], checkIfExists: true)
    germline_resource     = file(params.test_data['homo_sapiens']['genome']['gnomad_r2_1_1_21_vcf_gz'], checkIfExists: true)
    germline_resource_tbi = file(params.test_data['homo_sapiens']['genome']['gnomad_r2_1_1_21_vcf_gz_tbi'], checkIfExists: true)
    panel_of_normals      = file(params.test_data['homo_sapiens']['genome']['mills_and_1000g_indels_21_vcf_gz'], checkIfExists: true)
    panel_of_normals_tbi  = file(params.test_data['homo_sapiens']['genome']['mills_and_1000g_indels_21_vcf_gz_tbi'], checkIfExists: true)
    interval_file         = file(params.test_data['homo_sapiens']['genome']['genome_21_interval_list'], checkIfExists: true)
```

**Review by GCJMackenzie — CHANGES_REQUESTED** [review]
The gatk subworkflows were failing due to the recent changes in test datasets to prevent empty files being generated, reference files needed to be updated to use the chr21 references rather than the old chr22. These changes should get the tests working again.

**Comment by Emiller88** [comment]
Closing because of https://github.com/nf-core/modules/pull/2097 https://github.com/nf-core/modules/pull/2101 https://github.com/nf-core/modules/pull/2156

**Pull request closed without merging**